### PR TITLE
Use `normalizes` to prepare `Webhook#events` value

### PR DESCRIPTION
**Pull request by mjankowski**
Same deal as https://github.com/mastodon/mastodon/pull/27602 - left this out of initial PR post-upgrade. I think this one does not have the caching<>Marshall interaction issue the other one does.

### CODE REVIEW

File: app/models/webhook.rb [hunk 1]
```diff
@@ -33,11 +33,11 @@ class Webhook < ApplicationRecord
   validates :secret, presence: true, length: { minimum: 12 }
   validates :events, presence: true
 
-  validate :validate_events
+  validate :events_validation_error, if: :invalid_events?
```
Comment (ClearlyClaire) [inline app/models/webhook.rb:36]: Why this change?

Comment (mjankowski) [reply app/models/webhook.rb:36]: This change is not relevant to or required by the `normalizes` introduction. Made this change as a small refactor while adding the spec coverage here prior to adding the `normalizes`. Makes it easier to see in coverage report that both parts were actually run. Don't feel strongly about this if you hate it.

File: app/models/webhook.rb [hunk 2]
```diff
@@ -33,11 +33,11 @@ class Webhook < ApplicationRecord
...
   validate :validate_permissions
   validate :validate_template
 
-  before_validation :strip_events
+  normalizes :events, with: ->(events) { events.filter_map(&:strip).filter_map(&:presence) }
```
Comment (ClearlyClaire) [inline app/models/webhook.rb:40]: Why this change? Also it seems to run counter to #27744

Comment (mjankowski) [reply app/models/webhook.rb]: Yes it does ... I guess the rule is just for `map` and not `filter_map` as well. Either that or this was exluded from cop somehow. Either way, pushed a change to keep prior style.

**Review by ClearlyClaire — APPROVED** [review]
